statement.k

```diff
@@ -44,8 +44,108 @@ rule
     <k> #exeStmt(S:SimpleStatement;) => S; ...</k>
     <callStack> .List </callStack>
 
-rule
-    <k> #localFunctionCall(FN:Id, ARGS:Values) => functionCall(FN; ARGS) ... </k>     [structural]
+syntax ContractCellMap
+syntax Map ::= ContractCellMap
+
+
+//**************************************************************************//
+// I was planing to have the following rule for address conversion.
+// However, the behavior of the in_keys() funciton is not what I expected.
+// Thus, I have another work-around, which works, but may not be perfect.
+// Maybe someone can help me to check on this?
+
+/*
+rule [Address-Conversion]:
+    <k> #localFunctionCall(FN:Id, V:Value, .Values) => V ... </k>
+    <contracts> CTS </contracts>
+    requires (FN in_keys(CTS))
+*/
+
+rule [Address-Conversion]:
+    <k> #localFunctionCall(FN:Id, V:Value, .Values) => V ... </k>
+    <Contract>
+        <cname> FN </cname>
+        ...
+    </Contract>
```

Comment:
We should have a test for when the current contract has a method with the same name as another contract that it could be calling into.

Reply:
I have tried the following example in the Solidity Remix compiler.
```
contract Func {
    uint c;
    
    constructor() public {
        
    }
}

contract Test {
    uint a = 0;

    constructor () public {
        
    }
    
    function receiveEth(address addr_) public {
        Func funcTarget = Func(addr_);
    }
    
    function Func() public {
        uint temp;
    }
}
```
It ended up with a compilation error. To be more specific, it is a `TypeError` since `Func` in contract `Test` is a function type and cannot be used as the contract type `Func`. So, I guess we are safe to implement the rule like this.